File: lang/java/src/fun/vac/algorithms/number_theory/Coprimality.java
package fun.vac.algorithms.number_theory;

public final class Coprimality {

    private Coprimality() {}

    public static boolean reduceToBinaryGCD(long m, long n) {
        return GreatestCommonDivisor.iterativeBinaryGCD(m, n) == 1;
    }

    public static boolean reduceToEuclidean(long m, long n) {
        return GreatestCommonDivisor.iterativeEuclidean(m, n) == 1;
    }
}

File: lang/php/src/fun/vac/algorithms/number_theory/Coprimality.php
<?php
    namespace fun\vac\algorithms\number_theory\Coprimality;

    require_once "fun/vac/algorithms/number_theory/GreatestCommonDivisor.php";

    use fun\vac\algorithms\number_theory\GreatestCommonDivisor;

    function reduceToBinaryGCD(int $m, int $n): bool {
        return GreatestCommonDivisor\iterativeBinaryGCD($m, $n) == 1;
    }

    function reduceToEuclidean(int $m, int $n): bool {
        return GreatestCommonDivisor\iterativeEuclidean($m, $n) == 1;
    }
?>
